### PR TITLE
Added tests for multiple topics, created date, updated date, and stars.

### PR DESCRIPTION
**Pull request by arcanearronax**
Added additional tests for multiple topics, created date, update date, and star count. The tests do fail occasionally, but this appears to be an issue in main as well. This will require a different commit to resolve.

### CODE REVIEW

**Comment by arcanearronax** [comment]
Added additional tests for multiple topics, updated date, add date, and star count. Modified a few comparisons to None to use "is None" rather than "== None", as advised by pylint.

**Comment by hedythedev** [comment]
Thanks @arcanearronax, before your latest commit I think that the tests failed because we used `repo[‘pushed_at’]` Instead of updated_at

File: tests/test_search.py [hunk 1]
```diff
@@ -24,7 +26,89 @@ def test_search_topic():
...
+    for repo in repos:
+        assert repo["stargazers_count"] >= 0
+        assert repo["watchers_count"] >= 0
+        assert repo["forks_count"] >= 0
+        assert (repo["description"] is None) or repo["description"]
+        assert repo["full_name"].count("/") >= 1
+        assert repo["html_url"] == "https://github.com/" + repo["full_name"]
+
+        # Need to account for min and max created dates
+        repo_created_date = datetime.strptime(
```
Comment (hedythedev) [inline tests/test_search.py:67]: I think you could try making this what it was previously, but use pushed_at instead

**Comment by arcanearronax** [comment]
I'll need to take a second look at that since I thought I verified the data point. Part of the issue I was running into was the unauthorized request limit since I was running into failures after repeated runs.

Once I'm back at my computer I'll open a bug to handle unauthorized requests, since those seem to result in unhandled key errors.

**Comment by hedythedev** [comment]
@allcontributors please add @arcanearronax for tests and code

**Pull request merged**